File: C++/Tri_selection.cpp
#include <iostream>

void swap(int &a, int &b) {
    int temp = a;
    a = b;
    b = temp;
}

void selectionSort(int arr[], int size) {
    for (int i = 0; i < size - 1; i++) {
        // Trouver l'indice du minimum dans le tableau non trié
        int minIndex = i;
        for (int j = i + 1; j < size; j++) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }

        // Échanger l'élément minimum avec l'élément à la position i
        swap(arr[i], arr[minIndex]);
    }
}

//MAIN
int main() {
    int arr[] = {64, 25, 12, 22, 11};
    int size = sizeof(arr) / sizeof(arr[0]);

    std::cout << "Tableau avant le tri par sélection : ";
    for (int i = 0; i < size; i++) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    selectionSort(arr, size);

    std::cout << "Tableau après le tri par sélection : ";
    for (int i = 0; i < size; i++) {
        std::cout << arr[i] << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: Java/Tri_selection.java
public class Tri_selection {

    public static void triSelection(int[] tableau) {
        int n = tableau.length;

        for (int i = 0; i < n - 1; i++) {
            int indiceDuMinimum = i;

            // Recherche de l'indice du minimum dans le reste du tableau
            for (int j = i + 1; j < n; j++) {
                if (tableau[j] < tableau[indiceDuMinimum]) {
                    indiceDuMinimum = j;
                }
            }

            // Échange du minimum trouvé avec l'élément actuel
            int temp = tableau[i];
            tableau[i] = tableau[indiceDuMinimum];
            tableau[indiceDuMinimum] = temp;
        }
    }

    public static void main(String[] args) {
        int[] tableau = {64, 25, 12, 22, 11};
        triSelection(tableau);

        System.out.println("Tableau trié :");
        for (int valeur : tableau) {
            System.out.print(valeur + " ");
        }
    }
}
